sampleExtension/background.js

```diff
@@ -0,0 +1,121 @@
+    addBrowserActionListener();
+};
+
+// This function receives the blacklist from the sync storage.
+updateBlockedSites = function(callback){
+    chrome.storage.sync.get("tds_blacklist", function (items) {
+        if (!chrome.runtime.error) {
+            blockedSites = items.tds_blacklist;
+            return callback();
+        }
+    });
+};
+
+// This function adds one url to the blacklist
+addToBlockedSites = function(urlToAdd) {
+    var formatUrl = urlToAdd.split(['//'])[1];
+    formatUrl = '*://' + formatUrl + '*';
+    blockedSites.push(formatUrl);
+    chrome.storage.sync.set({"tds_blacklist" : blockedSites }, function() {
+        if(chrome.runtime.error) {
+            console.log("Runtime error.");
+        }
+    });
+};
+
+/* --------------- ------ Listener functions ------ ---------------*/
+
+replaceListener = function() {
+    chrome.webRequest.onBeforeRequest.removeListener(intercept);
+    addWebRequestListener();
+};
+
+addWebRequestListener = function() {
+    if(blockedSites != null && blockedSites.length > 0) {
+        chrome.webRequest.onBeforeRequest.addListener(
+                intercept,
+                {
+                    //Url's to be intercepted
+                    urls: blockedSites,
+                    types: ["main_frame"]
+            },
+            ["blocking"]
+        );
+    }
+};
+
+intercept = function() {
+    incrementInterceptionCounter();
+    return {redirectUrl: "https://zeeguu.herokuapp.com/getex"};
+    //return {redirectUrl: "https://zeeguu.herokuapp.com/getex"};
+};
+
+/* --------------- ------ ------------------ ------ ---------------*/
+
+addBrowserActionListener = function() {
+    chrome.browserAction.onClicked.addListener(function () {
+        chrome.tabs.query({active: true, currentWindow: true}, function (arrayOfTabs) {
+            // since only one tab should be active and in the current window at once
+            // the return variable should only have one entry
+            var activeTab = arrayOfTabs[0];
+            var activeTabUrl = activeTab.url;
+            addToBlockedSites(activeTabUrl);
+
+        });
+    });
+};
+
+/* --------------- ------ Statistics functions ------ ---------------*/
+
+incrementInterceptionCounter = function() {
+    chrome.storage.sync.get("tds_interceptCounter", function(output) {
+        var counter = output.tds_interceptCounter;
+        counter++;
+        chrome.storage.sync.set({"tds_interceptCounter": counter}, function () {
+            if (chrome.runtime.error) {
+                console.log("Runtime error.");
+            }
+        });
+    });
+};
+
+
+/* --------------- ---- Run upon installation ---- ---------------*/
+
+chrome.runtime.onInstalled.addListener(function() {
+    chrome.storage.sync.get("tds_blacklist", function(output) {
+        if (output.tds_blacklist == null) {
+            chrome.storage.sync.set({"tds_blacklist": []}, function () {
+                if (chrome.runtime.error) {
```

Comment:
Given the frequency with which you handle chrome runtime errors, wouldn't make sense to make a checkForRunTimeError() or something like it. If you ever decide to handle the runtime error not by printing to the console but by doing something else you only have to implement that in one place.